rdkit/Chem/MolStandardize/charge.py

```diff
@@ -314,13 +313,16 @@ def uncharge(self, mol):
                     log.info('Removed negative charge')
         else:
             #
-            for atom in [mol.GetAtomWithIdx(x) for x in n]:
+            negativeAtomIdxs = [mol.GetAtomWithIdx(x) for x in n]
+            for atom in negativeAtomIdxs:
                 while atom.GetFormalCharge() < 0:
                     atom.SetNumExplicitHs(atom.GetNumExplicitHs() + 1)
                     atom.SetFormalCharge(atom.GetFormalCharge() + 1)
                     log.info('Removed negative charge')
+        
         # Neutralize positive charges
-        for atom in [mol.GetAtomWithIdx(x) for x in p]:
+        positiveAtomIdxs = [mol.GetAtomWithIdx(x) for x in p]
+        for atom in positiveAtomIdxs:
```

Comment:
The definition of the list is so short that I don't see a benefit of separating it from the for loop in both cases. It also gives the impression that it could be used later on in the code.